did-utils/src/methods/did_peer/util.rs

```diff
@@ -84,13 +83,71 @@ fn abbrv_service(value: &mut Value, abbrv_key: &dyn Fn(&str) -> String, abbrv_va
     }
 }
 
+pub(super) fn validate_input_document(diddoc: &DIDDocument) -> Result<(), DIDPeerMethodError> {
+    // The document must not be empty
+    if diddoc.verification_method.is_none()
+        && diddoc.authentication.is_none()
+        && diddoc.assertion_method.is_none()
+        && diddoc.key_agreement.is_none()
+        && diddoc.capability_delegation.is_none()
+        && diddoc.capability_invocation.is_none()
+        && diddoc.service.is_none()
+    {
+        return Err(DIDPeerMethodError::InvalidStoredVariant);
+    }
+
+    // They must not be an id field at the root
+    // All id field within the document must be relative
+    // All references to ressources must be relative
+    if !diddoc.id.is_empty()
+        || !are_all_ids_and_references_relative(diddoc) {
+        return Err(DIDPeerMethodError::InvalidStoredVariant);
+    }
+    Ok(())
+}
```

Comment:
You seem to disregard the last check in the specs. Can you comment in the code why?
> For verification methods, the controller MUST be omitted if the controller is the document owner. If it is controlled by a DID other than the owner of the document, it MUST be included.

Reply:
This is a mistake. Will add the code for that tomorrow.

Reply:
I had a similar reasoning, that's why I recommend to add a comment explaining that in the code.

However, it is possible the specs have another reasoning, otherwise I don't see the point of adding that rule.